### DISCUSSION TITLE
Using mujoco.elasticity.solid and attaching two elastic objects to one body

### DISCUSSION THREAD

**Discussion by JedMuff** [google-deepmind/mujoco#2230] [Asking for Help] [answered]
### Intro

Hi!
I am a phd researcher at Vrije Universiteit Amsterdam, I use MuJoCo for my research on Evolutionary Robotics and I am currently trying out mujocos elasticity plugins but I am missing some information. 

### My setup

Python 3.12.3, Mujoco 3.2.1, Ubuntu 22.04.4 LTS

### My question

I am using the mujoco.elasticity.solid plugin. Firstly I notice how this elasticity plugin is not in the latest version of mujoco 3.2.4. My first question is why is that so? Should I be using mujoco.elasticity.shell? 

My second question is I am trying to attach two solid elastic cuboids (see code) to a solid cube and I get the following error: 
`    m = mujoco.MjModel.from_xml_string(XML, ASSETS)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Error: engine error: Body 2 does not have plugin instance 0`

I don't know what is causing this. Can someone help?

### Minimal model and/or code that explain my question

<mujoco model="scene">
  <compiler angle="radian" inertiafromgeom="true"/>
  <option timestep="0.001" gravity="0 0 -9.81" integrator="implicitfast" solver="CG"/>
  <extension>
    <plugin plugin="mujoco.elasticity.solid"/>
  </extension>
  <asset>
    <texture name="texgeom" type="cube" builtin="flat" rgb1="0.8 0.6 0.4" rgb2="0.8 0.6 0.4" mark="cross" markrgb="1 1 1" random="0.01" width="127" height="1278"/>
    <texture name="texplane" type="2d" builtin="checker" rgb1="0.137 0 0.275" rgb2="0.2 0 0.4" width="100" height="100"/>
    <material name="MatPlane" texture="texplane" texrepeat="60 60" specular="1" shininess="1" reflectance="0.5"/>
    <material name="geom" texture="texgeom" texuniform="true"/>
  </asset>
  <worldbody>
    <light name="light_0" directional="true" castshadow="false" pos="0 0 1.3" dir="0 0 -1.3" cutoff="100" exponent="1" diffuse="1 1 1" specular="0.1 0.1 0.1"/>
    <geom name="floor" type="plane" conaffinity="1" condim="3" size="40 40 40" material="MatPlane" rgba="0.8 0.9 0.8 1" friction="0.7 0.1 0.1" pos="0 0 0"/>
    <body name="root" pos="0 0 0.75">
      <camera name="track" mode="trackcom" pos="0 -3 -0.25" xyaxes="1 0 0 0 0 1"/>
      <freejoint name="root"/>
      <body name="Core" pos="0.089 0 1" quat="1 0 0 0">
        <geom name="Core" type="box" size="0.089 0.089 0.0603" rgba="255 0.25 0.25 1" friction="1 0.5 0.5"/>
        
        <flexcomp name="fr_flipper" type="grid" count="5 2 2" spacing="0.063 0.063 0.0603" radius="0" mass="5" rgba="255 5 0 1" pos="0.2147725 0 0" quat="1 0 0 0" dim="3">
          <contact condim="3" solref="0.01 1" solimp="0.95 0.98 0.0001" selfcollide="none"/>
          <plugin plugin="mujoco.elasticity.solid">
            <config key="poisson" value="0.9"/>
            <config key="young" value="5e-1"/>
          </plugin>
          <edge equality="true" damping="1"/>
          <pin id="1 2 3 4"/>
        </flexcomp>
        
        <flexcomp name="bl_flipper" type="grid" count="5 2 2" spacing="0.063 0.063 0.0603" radius="0" mass="5" rgba="255 5 0 1" pos="0.0 0.2147725 0" quat="0.70710678118654757 0 0 0.70710678118654746" dim="3">
          <contact condim="3" solref="0.01 1" solimp="0.95 0.99 0.0001" selfcollide="none"/>
          <plugin plugin="mujoco.elasticity.solid">
            <config key="poisson" value="0.9"/>
            <config key="young" value="5e-1"/>
          </plugin>
          <edge equality="true" damping="1"/>
          <pin id="1 2 3 4"/>
        </flexcomp>
      </body>
    </body>
  </worldbody>
</mujoco>

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by JedMuff** [comment]
...Well, almost 30mins later I found a solution to the bug 🤦. If you create a new body for each soft part it works. Does not answer my first question though. Also I wonder if you can connect two rigid bodies to one elastic solid. I tried but doesn't seem to work. I think this might be a limitation.

**Comment by quagla** [comment, marked as answer]
Hi the answer to the first question is that the solid plugin has now moved to the engine so you don't need a plugin at all :) see https://github.com/google-deepmind/mujoco/blob/main/model/flex/floppy.xml#L33

**Reply by JedMuff** [reply]
Oh thats really useful to know thank you!
